src/util/InteractionController.ts

```diff
@@ -0,0 +1,40 @@
+import { uuid } from 'uuidv4';
+import { ensureLeadingSlash, ensureTrailingSlash, trimTrailingSlashes } from './Util';
+import { LINK_TYPE_LDP_BC, LINK_TYPE_LDPC } from './LinkTypes';
+
+export class InteractionController {
+  /**
+   * Check whether a new container or a resource should be created based on the given parameters.
+   * @param slug - Incoming slug header.
+   * @param link - Incoming link header.
+   */
+  public isContainer(slug?: string, link?: Set<string>): boolean {
+    if (!slug) {
+      return link !== undefined && (link.has(LINK_TYPE_LDPC) || link.has(LINK_TYPE_LDP_BC));
+    }
+    if (slug.endsWith('/')) {
+      return !link || link.has(LINK_TYPE_LDPC) || link.has(LINK_TYPE_LDP_BC);
+    }
+    return link !== undefined && (link.has(LINK_TYPE_LDPC) || link.has(LINK_TYPE_LDP_BC));
```

Comment:
Would combine this with the first check since the results are identical (so change first if statement to `!slug || !slug.endsWith('/')`)